extensions/amp-video/0.1/amp-video.js

```diff
@@ -397,175 +511,147 @@ class AmpVideo extends AMP.BaseElement {
     return false;
   }
 
-
   /**
+   * @param {boolean} muted
    * @private
    */
-  installEventHandlers_() {
-    const video = dev().assertElement(this.video_);
-
-    this.unlisteners_.push(this.forwardEvents(
-        [VideoEvents.PLAYING, VideoEvents.PAUSE, VideoEvents.ENDED], video));
-
-    this.unlisteners_.push(listen(video, 'volumechange', () => {
-      if (this.muted_ != this.video_.muted) {
-        this.muted_ = this.video_.muted;
-        const evt = this.muted_ ? VideoEvents.MUTED : VideoEvents.UNMUTED;
-        this.element.dispatchCustomEvent(evt);
-      }
-    }));
-  }
-
-  /** @private */
-  uninstallEventHandlers_() {
-    while (this.unlisteners_.length) {
-      this.unlisteners_.pop().call();
+  setMuted_(muted) {
+    if (this.muted_ == muted) {
+      return;
     }
-  }
-
-  /**
-   * Resets the component if the underlying <video> was changed.
-   * This should only be used in cases when a higher-level component manages
-   * this element's DOM.
-   */
-  resetOnDomChange() {
-    this.video_ = dev().assertElement(
-        childElementByTag(this.element, 'video'),
-        'Tried to reset amp-video without an underlying <video>.');
-
-    this.uninstallEventHandlers_();
-    this.installEventHandlers_();
+    const evt = muted ? VideoEvents.MUTED : VideoEvents.UNMUTED;
+    this.muted_ = muted;
+    this.element.dispatchCustomEvent(evt);
   }
 
   /** @override */
   pauseCallback() {
-    if (this.video_) {
-      this.video_.pause();
-    }
+    this.getMixin_().pause();
   }
 
   /** @private */
   isVideoSupported_() {
-    return !!this.video_.play;
+    // TODO(alanorozco): Agh.
+    return true;
+    // return this.getMixin_().isVideoSupported();
   }
 
   // VideoInterface Implementation. See ../src/video-interface.VideoInterface
 
-  /**
-   * @override
-   */
+  /** @override */
+  preimplementsMediaSessionAPI() {
+    return false;
+  }
+
+  /** @override */
   supportsPlatform() {
     return this.isVideoSupported_();
   }
 
-  /**
-   * @override
-   */
+  /** @override */
   isInteractive() {
     return this.element.hasAttribute('controls');
   }
 
-  /**
-   * @override
-   */
+  /** @override */
   play(unusedIsAutoplay) {
-    const ret = this.video_.play();
-
-    if (ret && ret.catch) {
-      ret.catch(() => {
-        // Empty catch to prevent useless unhandled promise rejection logging.
-        // Play can fail for many reasons such as video getting paused before
-        // play() is finished.
-        // We use events to know the state of the video and do not care about
-        // the success or failure of the play()'s returned promise.
-      });
+    if (this.playbackMixin_ === null) {
+      return;
     }
+    dev().assert(this.playbackMixin_).play();
```

Comment:
The assert looks redundant with the guard clause above. Are both necessary?

Reply:
Looks like it wasn't necessary, thanks!